src/Polysemy/Internal.hs

```diff
@@ -540,6 +550,32 @@ subsumeUsing pr =
     go
 {-# INLINE subsumeUsing #-}
 
+------------------------------------------------------------------------------
+-- | Introduce a set of effects into 'Sem' at the index @i@, before the effect
+-- that previously occupied that position. This is intended to be used with a
+-- type application:
+--
+-- @
+-- let
+--   sem1 :: Sem [e1, e2, e3, e4, e5] a
+--   sem1 = insertAt @2 @[e3, e4] (sem0 :: Sem [e1, e2, e5] a)
+-- @
+--
+-- @since TODO
+insertAt
+  :: forall index inserted head oldTail tail old full a
+   . ( ListOfLength index head
+     , WhenStuck index InsertAtUnprovidedIndex
+     , old ~ Append head oldTail
+     , tail ~ Append inserted oldTail
+     , full ~ Append head tail
+     , InsertAtIndex index head tail oldTail full inserted)
```

Comment:
Good lord... I know there's not much we can do to shorten this type signature, but still.

Reply:
well those are basically local type binders…and an error message. Effectively it's just two constraints :upside_down_face: